backend/build_docs.py

```diff
@@ -0,0 +1,210 @@
+                f'{md_orfs_docs}/{file}', f'{cur_dir}/data/markdown/ORFS_docs/{file}'
+            )
+
+    for file in os.listdir(f'{md_orfs_docs}/user'):
+        if file.endswith('.md'):
+            if 'build' in file.lower():
+                copyfile(
+                    f'{md_orfs_docs}/user/{file}',
+                    f'{cur_dir}/data/markdown/ORFS_docs/installation/{file}',
+                )
+            else:
+                copyfile(
+                    f'{md_orfs_docs}/user/{file}',
+                    f'{cur_dir}/data/markdown/ORFS_docs/{file}',
+                )
+
+    print('Finished building ORFS docs.')
+
+    return
+
+
+def build_manpages() -> None:
+    print('Starting manpages build...')
+
+    # check if pandoc is installed, if not error out.
+    res = subprocess.run('pandoc --version', shell=True, capture_output=True)
+    if res.returncode != 0:
+        print('Pandoc is not installed. Please install it.')
+        sys.exit(1)
+    print('Pandoc is installed.')
+
+    # generate manpages
+    command = '../../etc/find_messages.py > messages.txt'
+    for module in os.listdir(os.path.join(cur_dir, 'OpenROAD/src')):
+        path = os.path.join(cur_dir, 'OpenROAD/src', module)
+        if not os.path.isdir(path):
+            continue
+        print('Processing module:', module)
+        os.chdir(path)
+        res = subprocess.run(command, shell=True, capture_output=True)
+        if res.returncode != 0:
+            print(
+                f"Error in finding messages for {module}: {res.stderr.decode('utf-8')}"
+            )
+            continue
+    os.chdir(os.path.join(cur_dir, 'OpenROAD/docs'))
+    num_cores = os.cpu_count()
+    command = f'make clean && make preprocess && make -j{num_cores}'
+    res = subprocess.run(command, shell=True, capture_output=True)
+    print('Finished building manpages.')
+
+    # copy folder contents to data/markdown/manpages
+    src_dir = os.path.join(cur_dir, 'OpenROAD/docs/md')
+    dest_dir = os.path.join(cur_dir, 'data/markdown/manpages')
+    shutil.copytree(src_dir, dest_dir, dirs_exist_ok=True)
+    print('Copied manpages to data/markdown/manpages.')
+
+    print('Finished building manpages.')
+
+    return
+
+
+def get_opensta_docs() -> None:
+    url = 'https://github.com/The-OpenROAD-Project/OpenSTA/raw/1c7f022cd0a02ce71d047aa3dbb64e924b6efbd5/doc/OpenSTA.pdf'
+
+    response = requests.get(url)
+
+    save_path = 'data/pdf/OpenSTA/OpenSTA_docs.pdf'
+    os.makedirs(os.path.dirname(save_path), exist_ok=True)
+
+    if response.status_code == 200:
+        with open(save_path, 'wb+') as file:
+            file.write(response.content)
+        print('OpenSTA docs downloaded successfully.')
+    else:
+        print('Failed to download file. Status code:', response.status_code)
+
+
+if __name__ == '__main__':
+    docs_paths = ['data']
+    purge_folders(folder_paths=docs_paths)
+
+    clone_repo(
+        url='https://github.com/The-OpenROAD-Project/OpenROAD.git',
```

Comment:
Lock a hash
This should work: https://stackoverflow.com/questions/75454944/how-to-clone-a-repository-from-a-specific-commit-id